lib/block-editor-settings.php

```diff
@@ -146,6 +146,47 @@ function gutenberg_get_block_editor_settings( $settings ) {
 			);
 	}
 
+	/*
+	 * Locale settings.
+	 *
+	 * Add user and site locale settings to block editor settings.
+	 * Because the current value of is_rtl() refers to the current locale,
+	 * we need to switch to the site locale to get the correct is_rtl() value for the site.
+	 *
+	 * Note: settings.isRTL is already available to determine the direction of the editor interface.
+	 *
+	 * Core backport notes:
+	 * - This could be added to block editor settings, e.g., $editor_settings array in /wp-includes/block-editor.php#L214
+	 * - Potential for abstracting this into a separate function in src/wp-includes/l10n.php, e.g., wp_get_site_locale() or something.
+	 *
+	 */
+	// Current user locale in the block editor.
+	$current_user_locale = get_user_locale();
+	$current_user_is_rtl = is_rtl();
+
+	// Current site locale.
+	$current_site_locale = get_locale();
+	$current_site_is_rtl = $current_user_is_rtl;
+
+	if ( $current_user_locale !== $current_site_locale ) {
+		$switched_locale = switch_to_locale( $current_site_locale );
+		if ( is_locale_switched() && $switched_locale ) {
+			$current_site_is_rtl = is_rtl();
+			restore_previous_locale();
+		}
+	}
+
+	$settings['locale'] = array(
+		'site' => array(
+			'lang'  => $current_site_locale,
+			'isRTL' => $current_site_is_rtl,
```

Comment:
@youknowriad What do you reckon about the locale data model here (and using it in block editor settings)?

Is there a better way to pass site and user locale data to the editor?

I looked at extending the site settings (`/wp/v2/settings`) endpoint, but it appears limited to site option values.

Reply:
> I looked at extending the site settings (/wp/v2/settings) endpoint, but it appears limited to site option values.

This would have been my preference personally as this is just a site option (whether it's stored as an option or not is not really important IMO). cc @TimothyBJacobs maybe

Reply:
Yeah, it'd be a neat place to house this stuff.

In that endpoint, we'd add a `is_rtl: true` or `dir: 'rtl'` or whatever, that corresponds to the **site locale**.

The only conceptual agony I was going through was that this value can't be updated in the POST like the others, so it isn't neat, but the world is messy so I can get over it. Keen to hear what Timothy thinks.

Here's the output of the GET request the record:

`await wp.data.resolveSelect( 'core' ).getSite()`

<details>

<summary>Respone</summary>

```json

{
    "title": "gutenberg",
    "description": "",
    "url": "http://localhost:8888",
    "email": "wordpress@example.com",
    "timezone": "",
    "date_format": "F j, Y",
    "time_format": "g:i a",
    "start_of_week": 1,
    "language": "ar",
    "use_smilies": true,
    "default_category": 1,
    "default_post_format": "0",
    "posts_per_page": 10,
    "show_on_front": "posts",
    "page_on_front": 0,
    "page_for_posts": 0,
    "default_ping_status": "open",
    "default_comment_status": "open",
    "site_logo": null,
    "site_icon": 0
}

```

</details>

Reply:
This is exactly what the index is really for. The `rest_index` filter can be used.

Reply:
Thanks for confirming @TimothyBJacobs I'll take a look at using the [filter](https://developer.wordpress.org/reference/hooks/rest_index/) 🙇🏻